contracts/src/core/EigenDADisperserRegistry.sol

```diff
@@ -22,11 +25,65 @@ contract EigenDADisperserRegistry is OwnableUpgradeable, EigenDADisperserRegistr
         _transferOwnership(_initialOwner);
     }
 
-    function setDisperserInfo(uint32 _disperserKey, DisperserInfo memory _disperserInfo) external onlyOwner {
+    /**
+     * @notice Register or update disperser info.
+     * @dev If the key is new, the sender must supply exactly REGISTRATION_DEPOSIT.
+     *      If already registered, only the original registrant may update and must supply an additional REGISTRATION_DEPOSIT.
+     *      All users, including the owner, must pay the full deposit.
+     */
+    function setDisperserInfo(uint32 _disperserKey, DisperserInfo memory _disperserInfo) external payable {
+        bool isOwner = msg.sender == owner();
+        
+        if (disperserRegistrants[_disperserKey] == address(0)) {
+            // New registration - all users must pay deposit
+            require(msg.value == REGISTRATION_DEPOSIT, "Deposit of 0.1 ether required for registration");
+            disperserRegistrants[_disperserKey] = msg.sender;
+            disperserDeposits[_disperserKey] = msg.value;
+        } else {
+            // Update – only the original registrant can update
+            require(disperserRegistrants[_disperserKey] == msg.sender, "Caller is not the registrant");
+            
+            // All users must pay additional deposit for updates
+            require(msg.value == REGISTRATION_DEPOSIT, "Additional deposit of 0.1 ether required for update");
+            disperserDeposits[_disperserKey] += msg.value;
+        }
+        
+        // Set disperser info
         disperserKeyToInfo[_disperserKey] = _disperserInfo;
+        
+        // If the sender is the owner, mark this disperser as owner-created
```

Comment:
What is the point of this marking? Is it used anywhere? If you want to give extra privileges offchain to an owner owned disperser key, maybe a better design would be to have an owner function that can mark keys as privileged.

Reply:
Yeah, the intend is to give extra privileges offchain. Your approach makes sense to me